File: utils/autostart.go
package utils

import (
	"fmt"
	"os"
	"os/exec"
	"os/user"
	"path/filepath"
)

func ToggleAutostart(enable bool) {
	autoDir := filepath.Join(os.Getenv("HOME"), ".config", "autostart")
	autoFile := filepath.Join(autoDir, "welcome.desktop")

	if _, err := os.Stat(autoDir); os.IsNotExist(err) {
		err := os.MkdirAll(autoDir, 0755)
		if err != nil {
			fmt.Println("Error creating directory:", err)
			return
		}
	}

	if enable {
		if _, err := os.Stat(autoFile); err == nil {
			fmt.Println("Autostart is already enabled")
			return
		}

		fmt.Println("Enabling autostart...")
		err := exec.Command("cp", "/usr/share/applications/welcome.desktop", autoFile).Run()
		if err != nil {
			fmt.Println("Error enabling autostart:", err)
			return
		}
		fmt.Println("Autostart enabled")
	} else {
		if _, err := os.Stat(autoFile); err != nil {
			fmt.Println("Autostart is already disabled")
			return
		}

		fmt.Println("Disabling autostart...")
		err := exec.Command("pkexec", "rm", autoFile).Run()
		if err != nil {
			fmt.Println("Error disabling autostart:", err)
			return
		}
		fmt.Println("Autostart disabled")
	}
}

func CheckFileExists() bool {
	usr, err := user.Current()
	if err != nil {
		return false
	}

	homeDir := usr.HomeDir
	filePath := filepath.Join(homeDir, ".config", "autostart", "welcome.desktop")

	_, err = os.Stat(filePath)
	return !os.IsNotExist(err)
}

File: utils/autostart.py
"""
Autostart management utilities
"""

import os
import subprocess
import shutil
from pathlib import Path


def toggle_autostart(enable):
    """
    Toggle autostart for the welcome application
    Args:
        enable: bool - True to enable autostart, False to disable
    """
    home = Path.home()
    autostart_dir = home / ".config" / "autostart"
    autostart_file = autostart_dir / "welcome.desktop"
    source_file = Path("/usr/share/applications/welcome.desktop")
    
    # Create autostart directory if it doesn't exist
    if not autostart_dir.exists():
        try:
            autostart_dir.mkdir(parents=True, mode=0o755)
        except Exception as e:
            print(f"Error creating directory: {e}")
            return
    
    if enable:
        # Enable autostart
        if autostart_file.exists():
            print("Autostart is already enabled")
            return
        
        print("Enabling autostart...")
        try:
            if source_file.exists():
                shutil.copy2(source_file, autostart_file)
                print("Autostart enabled")
            else:
                print(f"Source file {source_file} not found")
        except Exception as e:
            print(f"Error enabling autostart: {e}")
    else:
        # Disable autostart
        if not autostart_file.exists():
            print("Autostart is already disabled")
            return
        
        print("Disabling autostart...")
        try:
            # Use pkexec to remove file (in case permissions needed)
            subprocess.run(['pkexec', 'rm', str(autostart_file)], check=False)
            print("Autostart disabled")
        except Exception as e:
            print(f"Error disabling autostart: {e}")


def check_file_exists():
    """
    Check if autostart file exists
    Returns: bool - True if autostart is enabled, False otherwise
    """
    home = Path.home()
    autostart_file = home / ".config" / "autostart" / "welcome.desktop"
    return autostart_file.exists()